### PR TITLE
8266317: Vector API enhancements

### PR DESCRIPTION
**Pull request by PaulSandoz**
This PR contains API and implementation changes for [JEP-414 Vector API (Second Incubator)](https://openjdk.java.net/jeps/414), in preparation for when targeted.

Enhancements are made to the API for the support of operations on characters, such as for UTF-8 character decoding. Specifically, methods for loading/storing a `short` vector from/to a `char[]` array, and new vector comparison operators for unsigned comparisons with integral vectors. The x64 implementation is enhanced to supported unsigned comparisons.

Enhancements are made to the API for loading/storing a `byte` vector from/to a `boolean[]` array.

The testing of loads/stores can be expanded for scatter/gather, but before doing that i think some refactoring of the tests is required to reposition tests in the right classes. I would like to do that work after integration of the PR.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266317](https://bugs.openjdk.java.net/browse/JDK-8266317): Vector API enhancements

### Reviewers
 * [Jatin Bhateja](https://openjdk.java.net/census#jbhateja) (@jatin-bhateja - Committer) ⚠️ Review applies to de210a88c15b9d54b97305eb4822c09ff0cd6809
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**) ⚠️ Review applies to 93b093c1a84e620796d80d02648dfefaab5583b9

### Contributors
 * Paul Sandoz `<psandoz@openjdk.org>`
 * Sandhya Viswanathan `<sviswanathan@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3803/head:pull/3803` \
`$ git checkout pull/3803`

Update a local copy of the PR: \
`$ git checkout pull/3803` \
`$ git pull https://git.openjdk.java.net/jdk pull/3803/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3803`

View PR using the GUI difftool: \
`$ git pr show -t 3803`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3803.diff">https://git.openjdk.java.net/jdk/pull/3803.diff</a>

</details>

### CODE REVIEW

**Comment by PaulSandoz** [comment]
All tests pass for tier1,tier2,tier3 for build profiles linux-x64, linux-aarch64, macosx-x64, and windows-x64.

**Review by jatin-bhateja — APPROVED** [review]
Reviewed X86 backend changes for unsigned comparison.

File: src/hotspot/cpu/aarch64/aarch64.ad [hunk 1]
```diff
@@ -2437,6 +2437,10 @@ bool Matcher::supports_vector_variable_shifts(void) {
   return true;
 }
 
+bool Matcher::supports_unsigned_vector_comparison(int vlen, BasicType bt) {
```
Comment (iwanowww) [inline src/hotspot/cpu/aarch64/aarch64.ad:2440]: `Matcher::supports_vector_comparison_unsigned()` looks a better fit.

**Review by iwanowww — APPROVED** [review]
Looks good.

File: src/hotspot/cpu/aarch64/aarch64.ad (hunk 1 above)
Comment (PaulSandoz) [reply src/hotspot/cpu/aarch64/aarch64.ad]: I renamed the method (see commit).

**Review by iwanowww — APPROVED** [review]

**Pull request closed without merging**